test/framework/src/main/java/org/elasticsearch/transport/nio/SocketEventHandler.java

```diff
@@ -130,8 +131,8 @@ public void handleWrite(NioSocketChannel channel) throws IOException {
      * @param channel that was being written to
      * @param exception that occurred
      */
-    public void writeException(NioSocketChannel channel, Exception exception) {
-        logger.trace("failed to write to channel", exception);
+    void writeException(NioSocketChannel channel, Exception exception) {
+        logger.debug(new ParameterizedMessage("exception while writing to socket channel: {}", channel), exception);
```

Comment:
Since this is at the debug level, should you use a supplier here to avoid instantiating the parameterized message if not needed?